### PR TITLE
Issue #2339 - Chdir to the test-dir as part of the beforeFeature()

### PR DESCRIPTION
**Pull request by agajdosi**
Fix : #2339

### CODE REVIEW

**Comment by agajdosi** [comment]
@budhrg Because we recently got to the topic of using multiple MINISHIFT_HOMEs I am moving the default home dir in this PR one directory up. Proposed structure is:

```
directory minishift/out:
.
├── download (when we download addon to be installed)
├── test-results
│   └── integration_2018-4-27_10-01-23.log
└── test-run
    └── .minishift
    └── another-home (if we test different M home)
```

And the working directory of tests is in minishift/out, so we will have steps like:
```
Then JSON config file ".minishift/config/config.json" contains key "image-caching" with value matching "true"
When file from "https://raw.githubusercontent.com/minishift/minishift/master/addons/anyuid/anyuid.addon" is downloaded into location "download/anyuid"
And executing "minishift addons install download/anyuid" succeeds
```

WDYT?

**Comment by anjannath** [comment]
retest this please

**Comment by amitkrout** [comment]
@agajdosi TBH i don't understand much about the changes. I would be happy to do a BJN call to get more context. Probably on Monday will schedule a call. cc_ @budhrg @anjannath

**Comment by budhrg** [comment]
> TBH i don't understand much about the changes. I would be happy to do a BJN call to get more context. Probably on Monday will schedule a call. 

@amitkrout Sure

**Comment by anjannath** [comment]
retest this please

**Comment by amitkrout** [comment]
> I would really like to have one of the test case using different MINISHIFT_HOME directory in this PR

@budhrg Can you please add a usecase of test case using different MINISHIFT_HOME directory ?

**Comment by amitkrout** [comment]
retest this please

File: test/integration/features/cmd-addons.feature [hunk 1]
```diff
@@ -18,7 +18,7 @@ Feature: Addons command and its subcommands
   Scenario: Installing add-on from a folder
   Note: working directory when executing Minishift commands is: /test/integration.
```
Comment (amitkrout) [inline test/integration/features/cmd-addons.feature]: I think the working directory would be ```out/test-run```

File: test/integration/testsuite/prepare.go [hunk 2]
```diff
@@ -170,14 +171,14 @@ func runCommandsBeforeFeature(commands string, runner *util.MinishiftRunner) err
 }
 
 // copyOc copies the oc binary contained in directory structure of /<version>/<platform>/oc(.exe)
-// to cache/oc inside of the testDir.
-func copyOc(ocPath string, testDir string) error {
+// to cache/oc inside of the testDefaultHome.
+func copyOc(ocPath string, testDefaultHome string) error {
```
Comment (amitkrout) [inline test/integration/testsuite/prepare.go]: @agajdosi Did you verify it locally?

File: test/integration/features/cmd-addons.feature (hunk 1 above)
Comment (agajdosi) [reply test/integration/features/cmd-addons.feature]: @amitkrout Good catch, going to update now.

File: test/integration/testsuite/prepare.go (hunk 2 above)
Comment (agajdosi) [reply test/integration/testsuite/prepare.go]: @amitkrout yes, tried on Linux, works as expected

**Comment by agajdosi** [comment]
retest this please

**Comment by agajdosi** [comment]
retest this please

**Comment by agajdosi** [comment]
@minishift/minishift-dev Please take a look, PR is ready to be reviewed and merged probably.

File: docs/source/contributing/developing.adoc [hunk 3]
```diff
@@ -263,7 +263,7 @@ $ make integration MINISHIFT_BINARY=<path-to-custom-binary>
 ----
 
 TIMEOUT::
-The `TIMEOUT` parameter can be used to override the default timeout of `3600s`.
+The `TIMEOUT` parameter can be used to override the default timeout of `10800s`.
```
Comment (amitkrout) [inline docs/source/contributing/developing.adoc]: +1

File: test/integration/testsuite/prepare.go [hunk 4]
```diff
@@ -114,12 +114,13 @@ func setUp() string {
...
 
-	err = os.Setenv(constants.MiniShiftHomeEnv, testDir)
+	testDefaultHome := fmt.Sprintf("%s/.minishift", testDir)
+	err = os.Setenv(constants.MiniShiftHomeEnv, testDefaultHome)
 	if err != nil {
 		fmt.Printf("Error setting up environmental variable %v: %v\n", constants.MiniShiftHomeEnv, err)
 	}
 
-	return testDir
+	return testDir, testDefaultHome
```
Comment (LalatenduMohanty) [inline test/integration/testsuite/prepare.go]: @agajdosi  I am not able to recall what is the use of testDir. It is for temporarily storing the logs and artifacts, right?

Comment (LalatenduMohanty) [reply test/integration/testsuite/prepare.go]: @agajdosi  with your changes should  I expect the `.minishift` inside of `/tmp/minishift-tmp-test-dir-556220581`?

File: test/integration/testsuite/testsuite.go [hunk 5]
```diff
@@ -245,7 +246,7 @@ func FeatureContext(s *godog.Suite) {
 			os.Exit(1)
 		}
 
-		fmt.Println("Running Integration test in:", testDir)
+		fmt.Printf("Running Integration test in:%v.\nUsing MINISHIFT_HOME=%v\n", testDir, testDefaultHome)
```
Comment (LalatenduMohanty) [inline test/integration/testsuite/testsuite.go]: :+1:

**Pull request merged**